File: app/src/main/java/com/abelcht/n01f2smwc/smartwatch/communication/packages/ConfigurePackage.kt
package com.abelcht.n01f2smwc.smartwatch.communication.packages

class ConfigurePackage(bluetoothAddress: String) : SmartWatchPackage(
    0x32.toByte(), arrayOf(
        0x40.toByte(), 0xe2.toByte(), 0x01.toByte(), 0x00.toByte(), 0x00.toByte(), 0x00.toByte()
    ) + bluetoothAddress.split(":").map { it.toInt(16).toByte() }.toTypedArray()
)

File: python_framework/packages/ConfigurePackage.py
from typing import List

from packages.SmartWatchPackage import SmartWatchPackage


class ConfigurePackage(SmartWatchPackage):
    def __init__(self, bluetooth_address: List[int]):
        message_type = 0x32

        content = [0x40, 0xe2, 0x01, 0x00, 0x00, 0x00] + bluetooth_address
        super().__init__(message_type, content)
